config/default/manager/manager.yaml

```diff
@@ -42,6 +42,8 @@ spec:
         app: kubemacpool
         control-plane: mac-controller-manager
         controller-tools.k8s.io: "1.0"
+      annotations:
+        description: KubeMacPool manager pod manages MAC allocation to Pods and VMs in Kubernetes cluster
```

Comment:
I think you should capitalize manager at _"KubeMacPool manager"_ because its part of the component name "KubeMacPool Manager"
If you going to keep the _".. allocation to Pod and VM's.."_ I think you should change it to 
"manages Pods and VM's MAC addresses allocation"

Reply:
`"manages Pods' and VMs' MAC addresses allocation"` is not easily read IMO.
I think `KubeMacPool manages MAC allocation to Pods and VMs` brings the best of all of what you and Petr said.